RELEASE/scripts/autoscend.ash

```diff
@@ -2833,61 +2795,68 @@ boolean LX_freeCombats()
 	
 	if(my_adventures() < 2)
 	{
-		auto_log_warning("Too few adventures to safely automate free combats.", "red");
-		auto_log_warning("If we lose your last adv on a free combat the remaining free combats are wasted.", "red");
-		abort("Please perform the remaining free combats manually.");
+		auto_freeCombatsRemaining(true);
+		auto_log_warning("Too few adventures to safely automate free combats", "red");
+		auto_log_warning("If we lose your last adv on a free combat the remaining free combats are wasted", "red");
+		abort("Please perform the remaining free combats manually then run me again");
 	}
 
-	if((auto_my_path() != "Disguises Delimit") && neverendingPartyCombat())
+	if(neverendingPartyRemainingFreeFights() > 0)
 	{
-		return true;
+		if(powerlevel)
+		{
+			return neverendingPartyPowerlevel();
+		}
+		else
+		{
+			return neverendingPartyCombat();
+		}
 	}
+	
+	boolean adv_done = false;
 
 	if(!in_koe() && get_property("_machineTunnelsAdv").to_int() < 5 && canChangeToFamiliar($familiar[Machine Elf]))
 	{
-		if(get_property("auto_choice1119") != "")
-		{
-			set_property("choiceAdventure1119", get_property("auto_choice1119"));
-		}
-		set_property("auto_choice1119", get_property("choiceAdventure1119"));
-		set_property("choiceAdventure1119", 1);
-
+		backupSetting("choiceAdventure1119", 1);
 
 		familiar bjorn = my_bjorned_familiar();
 		if(bjorn == $familiar[Machine Elf])
 		{
 			handleBjornify($familiar[Grinning Turtle]);
 		}
 		handleFamiliar($familiar[Machine Elf]);
-		autoAdv(1, $location[The Deep Machine Tunnels]);
+		adv_done = autoAdv(1, $location[The Deep Machine Tunnels]);
 		if(bjorn == $familiar[Machine Elf])
 		{
 			handleBjornify(bjorn);
 		}
 
-		set_property("choiceAdventure1119", get_property("auto_choice1119"));
-		set_property("auto_choice1119", "");
+		restoreSetting("choiceAdventure1119");
 		handleFamiliar("item");
 		loopHandlerDelayAll();
-		return true;
+		return adv_done;
 	}
 
 	if(snojoFightAvailable())
 	{
-		handleFamiliar($familiar[Ms. Puck Man]);
-		autoAdv(1, $location[The X-32-F Combat Training Snowman]);
-		handleFamiliar("item");
-		if(get_property("_auto_digitizeDeskCounter").to_int() > 2)
-		{
-			set_property("_auto_digitizeDeskCounter", get_property("_auto_digitizeDeskCounter").to_int() - 1);
-		}
+		adv_done = autoAdv(1, $location[The X-32-F Combat Training Snowman]);
 		loopHandlerDelayAll();
-		return true;
+		return adv_done;
 	}
 
-	if(((my_level() < 13) || (get_property("auto_disregardInstantKarma").to_boolean())) && godLobsterCombat())
+	if(powerlevel)
 	{
-		return true;
+		return godLobsterCombat();
```

Comment:
just note that this is called in a loop as `if(!LX_freeCombats()) break;`, so this will silently block subsequent free fights from being used

I am starting to get a little worried that always returning the adventure success code is going to hide serious bugs.

Reply:
It was already doing that anyways though. Basically I made two changes here. first change was to replace `(my_level() < 13) || (get_property("auto_disregardInstantKarma").to_boolean()))` with `powerlevel`
giving the result of
`if(powerlevel && godLobsterCombat()) return true;'
I then changed that in a second step of
```
if(powerlevel)
  if(godLobsterCombat()) return true;
```
So yes if it fails it will skip the rest of them. But it was already doing this before, this is not a bug I introduced.... that all being said. just because I didn't introduce it right now doesn't mean I shouldn't fix it. I will go ahead and commit a fix to this. it will return true on success but continue onwards without aborting on a failure

Reply:
commit 78852d9